core/rawdb/database.go

```diff
@@ -256,12 +264,12 @@ func NewLevelDBDatabase(file string, cache int, handles int, namespace string, r
 
 // NewLevelDBDatabaseWithFreezer creates a persistent key-value database with a
 // freezer moving immutable chain segments into cold storage.
-func NewLevelDBDatabaseWithFreezer(file string, cache int, handles int, freezer string, namespace string, readonly bool) (ethdb.Database, error) {
-	kvdb, err := leveldb.New(file, cache, handles, namespace, readonly)
+func NewLevelDBDatabaseWithFreezer(file string, cache int, handles int, freezer string, namespace string, args ...bool) (ethdb.Database, error) {
```

Comment:
can you think twice about all functions with this `args` parameter

Reply:
The reason for introducing `args` param in is that I don't want the goroutine of frdb.freeze(db) running,  we can distinguish the offline block prune scenario if len(args) > 1. For this feature, there'll be many opening the same kvdb and two ancientdb, so running of this goroutine will bring unnecessary trouble even crash like db closed but freezer didn't at the same time. In addition, for geth offline case, it did not make sense that the item still move between db.

Reply:
I mean the `args` itself is meaningless, we don's know what the `args` are at the first glance

Reply:
Done